library/src/main/java/com/orm/SugarRecord.java

```diff
@@ -75,6 +75,14 @@
         return findById(type, Long.valueOf(id));
     }
 
+    public static <T> T first(Class<T>type){
+        return findById(type, 1);
+    }
+
+    public static <T> T last(Class<T>type){
+        return findById(type, count(type));
```

Comment:
As above, the item with an `id` equal to the total number of records might not be the last record if some objects were deleted from the database.

It'd be nice to actually capture both of these failing cases in your tests, and solve them in your implementations of `first` and `last`. If you need help writing those test cases, let me know and I'll throw in a commit!